src/main/java/org/candlepin/auth/CloudRegistrationAuth.java

```diff
@@ -267,23 +275,142 @@ public String generateRegistrationToken(Principal principal, CloudRegistrationIn
+        }
+
+        if (principal == null) {
+            throw new IllegalArgumentException("principal is null");
+        }
+
+        if (cloudRegInfo == null) {
+            throw new IllegalArgumentException("cloudRegistrationInfo is null");
+        }
+
+        CloudAuthenticationResult result = this.cloudRegistrationAdapter
+            .resolveCloudRegistrationDataV2(cloudRegInfo);
+
+        String cloudAccountId = result.getCloudAccountId();
+        if (cloudAccountId == null || cloudAccountId.isBlank()) {
+            String errmsg = this.i18nProvider.get()
+                .tr("cloud provider account could not be resolved");
+
+            throw new CloudRegistrationAuthorizationException(errmsg);
+        }
+
+        String cloudInstanceId = result.getCloudInstanceId();
+        if (cloudInstanceId == null || cloudInstanceId.isBlank()) {
+            String errmsg = this.i18nProvider.get()
+                .tr("cloud instance ID could not be resolved");
+
+            throw new CloudRegistrationAuthorizationException(errmsg);
+        }
+
+        String cloudProviderShortName = result.getCloudProviderShortName();
+        if (cloudProviderShortName == null || cloudProviderShortName.isBlank()) {
+            String errmsg = this.i18nProvider.get()
+                .tr("cloud provider short name could not be resolved");
+
+            throw new CloudRegistrationAuthorizationException(errmsg);
+        }
+
+        String productId = result.getProductId();
+        if (productId == null || productId.isBlank()) {
+            String errmsg = this.i18nProvider.get()
+                .tr("product id could not be resolved");
+
+            throw new CloudRegistrationAuthorizationException(errmsg);
+        }
+
+        String ownerKey = result.getOwnerKey();
+        boolean ownerAndPoolExists = false;
+        if (ownerKey != null && !ownerKey.isBlank() && result.isEntitled()) {
+            Owner owner = ownerCurator.getByKey(ownerKey);
+            if (owner != null) {
+                ownerAndPoolExists = !poolCurator.listByOwnerAndProduct(owner, productId).isEmpty();
+            }
+        }
+
+        String anonymousConsumerUuid = null;
+        if (ownerKey == null || ownerKey.isBlank() || !result.isEntitled() || !ownerAndPoolExists) {
+
+            // TODO: Schedule anonymous org creation job
+
+            AnonymousCloudConsumer anonConsumer = anonymousCloudConsumerCurator
+                .getByCloudInstanceId(cloudInstanceId);
+
+            if (anonConsumer == null) {
+                anonConsumer = new AnonymousCloudConsumer()
+                    .setCloudAccountId(cloudAccountId)
+                    .setCloudInstanceId(cloudInstanceId)
+                    .setProductId(productId)
+                    .setCloudProviderShortName(cloudProviderShortName);
+
+                anonConsumer = anonymousCloudConsumerCurator.create(anonConsumer);
+            }
```

Comment:
Something tricky here: In the case where the anonConsumer is not null here, it means the client already has an anon cloud consumer record, so it was already given an anonymous token beforehand. But the code goes on to generate a brand new anonymous token (with a renewed expiration date). This could potentially be abused for avoiding perpetually generating newer and newer anonymous tokens (and in extent, anonymous SCA certs). 
I wonder what should be done in this case.

One option would be to throw an exception at them that more or less says 'you already got an anonymous token, you aren't allowed another one'.

Another would be to send them a new anonymous token, but with a reduced expiration time, depending on the amount of time that has passed since the creation of the AnonCloudConsumer record (so if the original token is supposed to have 2 days of total duration until expiration, we would set the same expiration date on the new token: AnonCloudConsumer.created + 2 days). If 2 days have already passed, then we throw the exception. 

If they haven't properly registered within 2 days, something has gone really wrong, and they should open a customer case.

Reply:
A third option here, which I think would be the most elegant, is to have the jwt token stored at the time of the initial generation, and if a consumer for which we already have a record for tries to autoreg again, we just send back that original jwt token (which conveniently has the expiration date we want: 2 days after the original autoreg request).

The disadvantage is obviously that we would have to store the token, which takes extra cycles and DB storage for all autoreg requests. And I don't think this is a good trade, since this would only be useful for handling this edge case of someone doing stupid things using a custom script.

Reply:
I think I prefer the second option here. Any thoughts on it? @JoshAlbrecht23 @Ceiu

Reply:
Adding note here: We need feedback from PM before deciding the best way forward, but for now we leave the code as is (generate new token every time, with renewed expiration time). If that feedback arrives after this PR is merged, then it will be worked on in a separate task